ynput/OpenPype
# Global: Validate unique subsets

## Brief description

This implements a global validator whether the current publish will publish unique subsets (per asset).

This fixes #4330 

## Additional info

I've removed seemingly equivalent ones from Maya, Fusion and Photoshop - please double check.
There also appeared to be something relatively similar for Maya: 

It would be good to know whether there are any cases where this logic doesn't make sense. Would we ever allow a single publish have multiple instances set to the same asset+subset. Can we think of a reason?

## Testing notes:

I've only done a quick test in Maya.

1. Create some instances that publish to the same asset and subset names. (This should be **invalidated**!)
2. Create some instances that publish to different asset but match in subset names. (This should be allowed)
3. Create some instances that publish to same asset but differ in subset names. (This should be allowed)
4. Create some instances that publish to the same asset and subset names **but** during publishing deactive/disable the clashing ones. (This should be allowed)
